src/services/starcoder_model.py

```
@@ -0,0 +1,97 @@
+            adapter_found = False
+            for name, module in model.named_modules():
+                # Kiểm tra xem module có phải là AdaLoRA module không
+                if hasattr(module, 'lora_A') or hasattr(module, 'lora_B'):
+                    adapter_found = True
+                    print(f"Found adapter module: {name}")
+
+                    # Cách truy cập đúng cho ParameterDict trong AdaLoRA
+                    if hasattr(module, 'lora_A'):
+                        # Trong AdaLoRA, lora_A là ParameterDict, cần kiểm tra keys
+                        print(f"  lora_A keys: {list(module.lora_A.keys())}")
+                        for key in module.lora_A.keys():
```

Ruff (0.8.2)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)